crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -75,6 +75,13 @@ pub(super) fn place_comment<'a>(
             handle_leading_class_with_decorators_comment(comment, class_def)
         }
         AnyNodeRef::StmtImportFrom(import_from) => handle_import_from_comment(comment, import_from),
+        AnyNodeRef::ExprConstant(_) => {
+            if let Some(AnyNodeRef::ExprFString(fstring)) = comment.enclosing_parent() {
+                CommentPlacement::dangling(fstring, comment)
```

Comment:
Could you instead check whether the comment is between two string constants? i'd like to avoid reintroducing `enclosing_parent` if possible.

Reply:
Relatedly, where is the comment currently attached and would it be possible to check for constant comment in string formatting itself?

Reply:
> Could you instead check whether the comment is between two string constants? i'd like to avoid reintroducing enclosing_parent if possible.

I would also like to avoid re-introducing enclosing_parent. However, the problem is that the comments fall into a `ConstantExpr` range and `following` and `preceding` are both `None`. 

> Relatedly, where is the comment currently attached and would it be possible to check for constant comment in string formatting itself?

The comments are attached to the `ExprConstant` without this change. I tried changing the string formatting to introduce a `AnyStringDanglingCommentsIter` that iterates over all dangling comments (includes all parts for an `FString`). However, I ran into problems because `leading_comments` and `trailing_comments` expect a `&[SourceComment]`. I tried collecting into a `Vec` but that doesn't work because the Vec's slice is of the type `&[&SourceComment]` (a slice of source comment references instead of source comments).

I then looked into changing `leading_comments` and `trailing_comments` to accept an iterator instead, but that complicates their implementations (and signatures) significantly. 

Lastly, I tried to add a `AnyString::parts` method that returns a `&[Expr]` slice. Unfortunately, this doesn't work for the same reasons as above AND has the additional issue that it's impossible to cast a `&ExprConstant` to a `&Expr`. 

I think changing `leading_` and `trailing_comments` is worth the effort if the current f-string formatting would be the final solution. But I don't think it is. I expect it to require significant re-work to support comments and nested parentheses inside of `FormattedValue`.

Reply:
The reason why I prefer re-introducing `enclosing_parent` is because rust will warn us when it is unused. Rust won't warn us if the `IntoIterator` variant of `leading_comments` is no longer needed. Meaning, that complexity likely remains forever.

Reply:
thanks for the explanation!